src/pal/src/thread/process.cpp

```diff
@@ -4389,25 +4390,24 @@ getPath(
         nextLen = strlen(lpNext);
         slashLen = (lpNext[nextLen-1] == '/') ? 0:1;
 
-        /* verify if the path fit in the OUT parameter */
-        if (slashLen + nextLen + strlen (lpFileName) >= iLen)
+        if (!lpPathFileName.Reserve(nextLen + lpFileNameString.GetCount()))
```

Comment:
I am still adding a 100 characters on every resize on the heap, so these smaller appends should be fine